lib/ImplicitDiscreteSolve/src/cache.jl

```diff
@@ -1,36 +1,66 @@
-mutable struct ImplicitDiscreteState{uType, pType, tType}
+struct ImplicitDiscreteState{uType, pType, tType}
     u::uType
     p::pType
-    t_next::tType
+    t::tType
 end
 
-mutable struct IDSolveCache{uType} <: OrdinaryDiffEqMutableCache
+mutable struct IDSolveCache{uType, cType, thetaType} <: OrdinaryDiffEqMutableCache
     u::uType
     uprev::uType
-    state::ImplicitDiscreteState
-    prob::Union{Nothing, SciMLBase.AbstractNonlinearProblem}
+    z::uType
+    nlcache::cType
+    Θks::thetaType
 end
 
 function alg_cache(alg::IDSolve, u, rate_prototype, ::Type{uEltypeNoUnits},
         ::Type{uBottomEltypeNoUnits}, ::Type{tTypeNoUnits}, uprev, uprev2, f, t,
         dt, reltol, p, calck,
         ::Val{true}) where {uEltypeNoUnits, uBottomEltypeNoUnits, tTypeNoUnits}
-    state = ImplicitDiscreteState(isnothing(u) ? nothing : zero(u), p, t)
-    IDSolveCache(u, uprev, state, nothing)
+    state = ImplicitDiscreteState(zero(u), p, t)
+    f_nl = (resid, u_next, p) -> f(resid, u_next, p.u, p.p, p.t)
```

Comment:
What is the reasoning here to include the current `u` in the signature of this function, but no information on dt?

Reply:
I guess my question is simply, why is $dt$ (or tprev) not a parameter, but $uprev$ is part of the function signature?